test.py

```diff
@@ -680,14 +680,15 @@ def set_summary(summary):
                         raise
                 set_summary("failed: {}".format(e))
             finally:
-                if self.is_new or self.is_equal_result is False:
-                    # Put a copy of the .reject file close to the .result file
-                    # so that it's easy to analyze the diff or overwrite .result
-                    # with .reject. Preserve the original .reject file: in
-                    # multiple modes the copy .reject file may be overwritten.
-                    shutil.copyfile(self.tmpfile, self.reject)
-                elif os.path.exists(self.tmpfile):
-                    pathlib.Path(self.tmpfile).unlink()
+                if os.path.exists(self.tmpfile):
+                    if self.is_executed_ok and (self.is_new or self.is_equal_result is False):
+                        # Put a copy of the .reject file close to the .result file
+                        # so that it's easy to analyze the diff or overwrite .result
+                        # with .reject. Preserve the original .reject file: in
+                        # multiple modes the copy .reject file may be overwritten.
+                        shutil.move(self.tmpfile, self.reject)
```

Comment:
The comment explains why this should be a copy, not a move.

Reply:
Unless I don't understand something. Is the original .reject file preserved? How?

Reply:
Also, since the purpose of the patch is to handle missing log file - why do you also replace the copy with a move? You didn't motivate this change in the commit message.

Reply:
>  Unless I don't understand something. Is the original .reject file preserved? How?

So IIUC `self.tmpfile` apparently is what the comment calls "the original .reject file". Thus the change to `move` is incompatible with the comment.